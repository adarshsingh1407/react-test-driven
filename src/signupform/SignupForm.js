import React, { Component, Fragment } from 'react';

class SignupForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
      name: 'Adarsh',
      username: ''
    };
  }
  handleChange = (name) => {
    this.setState({
      name
    });
  }
  handleSubmit = () => {
    const {name} = this.state;
    this.setState({
      username: name
    });
  }
  render() {
    const {name, username} = this.state;
    return (
      <Fragment>
        <div id="username">{username}</div>
        <input id="username-input" type="text" name="" value={name} onChange={(e) => this.handleChange(e.target.value)} />
        <button id="signupform-submit" type="submit" value="Submit" onClick={this.handleSubmit}>Submit</button>
      </Fragment>
    );
  }

}

export default SignupForm;
