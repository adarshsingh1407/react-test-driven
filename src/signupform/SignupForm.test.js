import React from 'react';
import {shallow} from 'enzyme';
import SignupForm from './SignupForm';

describe('SignupForm', () => {
  it('should render correctly in debug mode', () => {
    const component = shallow(<SignupForm />);
    expect(component).toMatchSnapshot();
  })
})
