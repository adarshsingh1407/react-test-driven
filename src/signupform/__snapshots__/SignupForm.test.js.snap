// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignupForm should render correctly in debug mode 1`] = `
<Fragment>
  <input
    name=""
    onChange={[Function]}
    type="text"
    value=""
  />
  <button
    onClick={[Function]}
    type="submit"
    value="Submit"
  >
    Submit
  </button>
</Fragment>
`;
